src/test/java/com/spotify/futures/FuturesExtraTest.java

```diff
@@ -571,4 +572,78 @@ public void testAsyncException6() throws Exception {
             });
     future.get();
   }
+
+  @Test
+  public void testWaitForNoInputs() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future);
+    assertSame(future, result);
+    assertTrue(result.isDone());
+
+  }
+
+  @Test
+  public void testWaitForOneInput() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1);
+
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test
+  public void testWaitForTwoInputs() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> other2 = Futures.immediateFuture("other2");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2);
+
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test
+  public void testWaitForTwoInputsOneFailure() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> other2 = Futures.immediateFailedFuture(new RuntimeException());
+    final ListenableFuture<String> other3 = Futures.immediateFuture("other2");
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2, other3);
+
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test
+  public void testWaitForNotImmediate() throws Exception {
+    final ListenableFuture<String> future = Futures.immediateFuture("hello");
+    final ListenableFuture<String> other1 = Futures.immediateFuture("other1");
+    final ListenableFuture<String> other2 = Futures.immediateFailedFuture(new RuntimeException());
+    final SettableFuture<String> other3 = SettableFuture.create();
+    final ListenableFuture<String> result = FuturesExtra.waitFor(future, other1, other2, other3);
+
+    assertFalse(result.isDone());
+    other3.set("foo");
+    assertTrue(result.isDone());
+    assertEquals("hello", result.get());
+
+  }
+
+  @Test(expected = ExecutionException.class)
+  public void testWaitForNotImmediateWithFailure() throws Exception {
```

Comment:
Wouldn't it be better semantics to have waitFor fail and report the exception if any of the waited for futures fail? It seems like you'd want to know if something fails. And if you don't, you can always use a fallback or something to ensure the other futures you're waiting for never fail.